lib/spack/spack/test/env.py

```diff
@@ -1509,3 +1509,52 @@ def test_cannot_use_double_percent_with_require(tmp_path, mutable_config):
     with ev.Environment(tmp_path) as e:
         with pytest.raises(spack.solver.asp.UnsatisfiableSpecError, match="failed to concretize"):
             e.concretize()
+
+
+@pytest.mark.parametrize(
+    "spack_yaml",
+    [
+        # Specs with reuse on
+        """
+spack:
+  specs:
+  - trilinos
+  - mpileaks
+  concretizer:
+    reuse: true
+""",
+        # Package with conditional dependency
+        """
+spack:
+  specs:
+  - ascent+adios2
+  - fftw+mpi
+""",
+        """
+spack:
+  specs:
+  - ascent~adios2
+  - fftw~mpi
+""",
+        """
+    spack:
+      specs:
+      - ascent+adios2
+      - fftw~mpi
+    """,
```

Comment:
```suggestion
""",
```

Reply:
Hmm, this was indented by `black` :thinking: